pkg/messageprocessors/javascript/javascript.go

```diff
@@ -59,45 +61,72 @@ var (
 )
 
 func wrapDownlinkEncoderScript(script string) string {
-	// Fallback to legacy Encoder() function for backwards compatibility with The Things Network Stack V2 payload functions.
+	// Fallback to Encoder() for backwards compatibility with The Things Network Stack V2 payload functions.
 	return fmt.Sprintf(`
 		%s
 
 		function main(input) {
+			const { data, fPort } = input;
 			if (typeof encodeDownlink === 'function') {
-				return encodeDownlink(input);
+				return encodeDownlink({ data, fPort });
 			}
 			return {
-				bytes: Encoder(input.data, input.fPort),
+				bytes: Encoder(data, fPort),
 				fPort: input.fPort
 			}
 		}
 	`, script)
 }
 
 // CompileDownlinkEncoder generates a downlink encoder from the provided script.
-func (h *host) CompileDownlinkEncoder(ctx context.Context, script string) (func(context.Context, *ttnpb.EndDeviceIdentifiers, *ttnpb.EndDeviceVersionIdentifiers, *ttnpb.ApplicationDownlink) error, error) {
+func (h *host) CompileDownlinkEncoder(
+	ctx context.Context, script string,
+) (
+	func(
+		context.Context,
+		*ttnpb.EndDeviceIdentifiers,
+		*ttnpb.EndDeviceVersionIdentifiers,
+		*ttnpb.ApplicationDownlink,
+	) error,
+	error,
+) {
 	defer trace.StartRegion(ctx, "compile downlink encoder").End()
 
 	run, err := h.engine.Compile(ctx, wrapDownlinkEncoderScript(script))
 	if err != nil {
 		return nil, err
 	}
 
-	return func(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers, version *ttnpb.EndDeviceVersionIdentifiers, msg *ttnpb.ApplicationDownlink) error {
-		return h.encodeDownlink(ctx, ids, version, msg, run)
+	return func(
+		ctx context.Context,
+		ids *ttnpb.EndDeviceIdentifiers,
+		version *ttnpb.EndDeviceVersionIdentifiers,
+		msg *ttnpb.ApplicationDownlink,
+	) error {
+		return h.encodeDownlink(ctx, msg, run)
 	}, nil
 }
 
 // EncodeDownlink encodes the message's DecodedPayload to FRMPayload using the given script.
-func (h *host) EncodeDownlink(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers, version *ttnpb.EndDeviceVersionIdentifiers, msg *ttnpb.ApplicationDownlink, script string) error {
+func (h *host) EncodeDownlink(
+	ctx context.Context,
+	_ *ttnpb.EndDeviceIdentifiers,
+	_ *ttnpb.EndDeviceVersionIdentifiers,
+	msg *ttnpb.ApplicationDownlink,
+	script string,
+) error {
 	run := func(ctx context.Context, fn string, params ...interface{}) (func(interface{}) error, error) {
 		return h.engine.Run(ctx, wrapDownlinkEncoderScript(script), fn, params...)
 	}
-	return h.encodeDownlink(ctx, ids, version, msg, run)
+	return h.encodeDownlink(ctx, msg, run)
 }
 
-func (h *host) encodeDownlink(ctx context.Context, ids *ttnpb.EndDeviceIdentifiers, version *ttnpb.EndDeviceVersionIdentifiers, msg *ttnpb.ApplicationDownlink, run func(context.Context, string, ...interface{}) (func(interface{}) error, error)) error {
+func (*host) encodeDownlink(
+	ctx context.Context,
+	msg *ttnpb.ApplicationDownlink,
+	run func(context.Context, string, ...interface{},
```

Comment:
Nit - Do you need the new line for the return values ? Can we do the parameters of `run` in the same style as we do in other places (so the list goes on its own line).

Reply:
No I wasn't paying attention here